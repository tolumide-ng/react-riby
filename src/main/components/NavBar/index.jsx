import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBar = () => (
  <div className="w-full shadow-md flex flex-row justify-between py-2 fixed bg-white">
    <p className="pl-4">React-Riby</p>
    <div data-testid="navbar" className="pr-4">
      <NavLink to="/" className="mr-4 focus:bg-gray-300 px-3 py-2">
        Home
      </NavLink>
      <NavLink to="/search" className="focus:bg-gray-300 px-3 py-2">
        Search
      </NavLink>
    </div>
  </div>
);

export default NavBar;
