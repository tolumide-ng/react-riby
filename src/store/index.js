import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './modules';

const middlewares = applyMiddleware(ReduxThunk);

const store = createStore(rootReducer, composeWithDevTools);

export default store;
