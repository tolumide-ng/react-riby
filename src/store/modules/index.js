import { combineReducers } from 'redux';
import allGifReducer from './allGifs/reducers';
import detailedGifReducer from './detailedGif/reducers';
import searchGifReducer from './searchGif/reducers';

const rootReducer = combineReducers({
  allGifReducer,
  detailedGifReducer,
  searchGifReducer,
});

export default rootReducer;
