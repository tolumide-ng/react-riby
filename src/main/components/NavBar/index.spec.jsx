/* eslint-disable react/jsx-props-no-spreading */
import { render, fireEvent, cleanup } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import NavBar from './index';
import store from '../../../store';

describe('Loader', () => {
  test('Should return the Navigation Bar', () => {
    const { asFragment, getByTestId, getByText } = render(
      <Provider store={store}>
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
    expect(getByTestId('navbar')).toBeTruthy();
    expect(getByText('React-Riby')).toBeTruthy();
  });
});
