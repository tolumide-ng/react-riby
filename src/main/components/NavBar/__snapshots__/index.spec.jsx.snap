// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should return the Navigation Bar 1`] = `[Function]`;
